mocks/tests/conftest.py

```diff
@@ -46,3 +46,27 @@ def custom_cosmology_data():
     custom_data['hubble_constant'] = 70.
     custom_data['T_CMB0'] = 2.5
     return custom_data
+
+
+@pytest.fixture
+def configuration():
+    config = dict()
+    config['galaxy_coordinates'] = '14h40m56.435s -60d53m48.3s'
+    config['observation_time'] = '2018-04-12T08:00'
+    config['data_path'] = 'sim_data/cl19.fits'
+    config['imager_filter'] = 'g'
+    config['mass_to_light_ratio'] = dict()
+    config['mass_to_light_ratio']['g'] = 5
+    config['mass_to_light_ratio']['r'] = 5
+    config['abs_mag_sun'] = dict()
+    config['abs_mag_sun']['g'] = 5.11
+    config['abs_mag_sun']['r'] = 4.68
+    config['abs_mag_sun']['i'] = 4.53
+    config['galaxy_distance'] = 10.
+    config['sim_pc_pixel'] = 170
+    config['particle_baryonic_mass_sim'] = 9.6031355
+    config['hubble_constant'] = 0.705
+    config['Omega_Lambda'] = 0.727
+    config['Omega_0'] = 0.273
+    config['redshift'] = 0
+    return config
```

Comment:
I did not want to load the configuration file for tests other than the one which is testing the `mocks.load_configuration` so I created this `fixture`. It is too long and that makes me think maybe I should try something else, so I will appreciate any advice about it.

Reply:
Why not use configuration files for all the tests? It would be simpler than this.

What you do is have a test fixture that loads the config and returns it (either here or in `tests/test_mocks.py`), then have a test use the fixture and check that it loaded OK, e.g.

```
from mock.utils import load_yaml_config

@pytest.fixture(scope='package')
def configuration():
    return load_yaml_config()
```

and

```
def test_load_configuration(configuration):
    assert configuration['data_path'] == 'sim_data/cl19.fits'
    assert configuration['galaxy_distance'] == 10.
￼    assert isinstance(configuration, dict)
```
The first test to use the `configuration` fixture will trigger the call to `load_yaml_config`, and if that fails with an exception then the exception will be passed up to the test, and it will fail (as it should). If it passes then the rest of the tests can run using the same `configuration` fixture to access the test config.

Reply:
I applied you suggestion, but is that alright that the function `utils.load_yaml_config` is tested twice? Once here and another time in `test_utils.py`